src/barretenberg_structures.rs

```diff
@@ -641,13 +648,91 @@ impl ConstraintSystem {
             buffer.extend(&constraint.to_bytes());
         }
 
+        // Serialize each Block constraint
+        let len = self.block_constraints.len() as u32;
+        buffer.extend_from_slice(&len.to_be_bytes());
+        for constraint in self.block_constraints.iter() {
+            buffer.extend(&constraint.to_bytes());
+        }
+
+        buffer
+    }
+}
+
+#[derive(Clone, Hash, Debug)]
+pub(crate) struct MemOpBarretenberg {
+    pub(crate) is_store: i8,
+    pub(crate) index: Constraint,
+    pub(crate) value: Constraint,
+}
+impl MemOpBarretenberg {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut buffer = Vec::new();
+
+        buffer.extend_from_slice(&self.is_store.to_be_bytes());
+        buffer.extend_from_slice(&self.index.to_bytes());
+        buffer.extend_from_slice(&self.value.to_bytes());
+
+        buffer
+    }
+}
+
+#[derive(Clone, Hash, Debug)]
+pub(crate) struct BlockConstraint {
+    pub(crate) init: Vec<Constraint>,
+    pub(crate) trace: Vec<MemOpBarretenberg>,
+    pub(crate) is_ram: i8,
+}
+
+impl BlockConstraint {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut buffer = Vec::new();
+
+        let len = self.init.len() as u32;
+        buffer.extend_from_slice(&len.to_be_bytes());
+        for value in self.init.iter() {
+            buffer.extend_from_slice(&value.to_bytes());
+        }
```

Comment:
Not a blocker to this PR but we should factor this logic out as it's used so much.